### ISSUE TITLE
Add Typescript types to AddSelect

### ISSUE THREAD

**Issue by matthewgallo** [carbon-design-system/ibm-products#5123]

**Comment by sstrubberg** [comment]
@makafsal I see you have moved this from Needs Triage to In Progress to Needs Review.

 Is this currently being worked on this sprint? Is there a PR attached to it?

**Comment by makafsal** [comment]
@sstrubberg Yeah, I've raised a PR against this, because the `AddSelect` component is related to SingleAddSelect, so I migrated it together.
